devicetypes/smartthings/orvibo-motion-sensor.src/Orvibo-Motion-Sensor.groovy

```diff
@@ -0,0 +1,167 @@
+		for (int i = 0; i <= 100; i += 11) {
+			status "battery ${i}%": "read attr - raw: 2E6D01000108210020C8, dni: 2E6D, endpoint: 01, cluster: 0001, size: 08, attrId: 0021, encoding: 20, value: ${i}"
+		}
+	}
+	preferences {
+		section {
+			image(name: 'educationalcontent', multiple: true, images: [
+					"http://cdn.device-gse.smartthings.com/Motion/Motion1.jpg",
+					"http://cdn.device-gse.smartthings.com/Motion/Motion2.jpg",
+					"http://cdn.device-gse.smartthings.com/Motion/Motion3.jpg"
+			])
+		}
+	}
+	tiles(scale: 2) {
+		multiAttributeTile(name: "motion", type: "generic", width: 6, height: 4) {
+			tileAttribute("device.motion", key: "PRIMARY_CONTROL") {
+				attributeState "active", label: 'motion', icon: "st.motion.motion.active", backgroundColor: "#00A0DC"
+				attributeState "inactive", label: 'no motion', icon: "st.motion.motion.inactive", backgroundColor: "#cccccc"
+			}
+		}
+		valueTile("battery", "device.battery", decoration: "flat", inactiveLabel: false, width: 2, height: 2) {
+			state "battery", label: '${currentValue}% battery', unit: ""
+		}
+		standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", action: "refresh.refresh", icon: "st.secondary.refresh"
+		}
+		main(["motion"])
+		details(["motion","battery", "refresh"])
+	}
+}
+def collectAttributes(Map descMap) {
+	def descMaps = new ArrayList<Map>()
+	descMaps.add(descMap)
+	if (descMap.additionalAttrs) {
+		descMaps.addAll(descMap.additionalAttrs)
+	}
+	return  descMaps
+}
+def stopMotion() {
+	log.debug "motion inactive"
+	sendEvent(getMotionResult('inactive'))
+}
+def installed(){
+	log.debug "installed"
+	return zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0021) +
+					zigbee.readAttribute(zigbee.IAS_ZONE_CLUSTER,zigbee.ATTRIBUTE_IAS_ZONE_STATUS)
+
+}
+def parse(String description) {
+	log.debug "description(): $description"
+	def map = zigbee.getEvent(description)
+	if(!map){
+		if (description?.startsWith('zone status')) {
+			map = parseIasMessage(description)
+			motionHandler(description);
+		} else {
+			map = batteyHandler(description);
+		}
+	}
+	log.debug "Parse returned $map"
+	def result = map ? createEvent(map) : [:]
+	if (description?.startsWith('enroll request')) {
+		List cmds = zigbee.enrollResponse()
+		log.debug "enroll response: ${cmds}"
+		result = cmds?.collect { new physicalgraph.device.HubAction(it) }
+	}
+	return result
+}
+def batteyHandler(String description){
+	def descMap = zigbee.parseDescriptionAsMap(description)
+    def map
+	if (descMap?.clusterInt == zigbee.POWER_CONFIGURATION_CLUSTER && descMap.commandInt != 0x07 && descMap.value) {
+		log.info "BATT METRICS - attr: ${descMap?.attrInt}, value: ${descMap?.value}, decValue: ${Integer.parseInt(descMap.value, 16)}, currPercent: ${device.currentState("battery")?.value}, device: ${device.getDataValue("manufacturer")} ${device.getDataValue("model")}"
+		List<Map> descMaps = collectAttributes(descMap)
```

Comment:
Refactor this part of code.  Check several things:
1. What is reported in descMap.attrInt  (if it is 0x0021 some of code is not necessary)
2. Is "collectAttributes()" required in that case. Does device reports anything in descMap.additionalAttrs?